src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/SqlException.cs

```diff
@@ -40,23 +40,29 @@ private SqlException(SerializationInfo si, StreamingContext sc) : base(si, sc)
             {
                 if ("ClientConnectionId" == siEntry.Name)
                 {
-                    _clientConnectionId = (Guid)siEntry.Value;
+                    _clientConnectionId = new Guid(siEntry.Value.ToString());
                     break;
                 }
             }
-
         }
 
         /// <include file='..\..\..\..\..\..\..\doc\snippets\Microsoft.Data.SqlClient\SqlException.xml' path='docs/members[@name="SqlException"]/GetObjectData/*' />
         override public void GetObjectData(SerializationInfo si, StreamingContext context)
         {
-            if (null == si)
+            base.GetObjectData(si, context);
+            si.AddValue("Errors", null); // Not specifying type to enable serialization of null value of non-serializable type
```

Comment:
This change was necessary as otherwise serialization wouldn't work.
Possible issue in .NET Framework source code.